### ISSUE TITLE
Any roadmap for IPv6?

### ISSUE THREAD

**Issue by theotherguy2175** [ckabalan/visualsubnetcalc#3]

**Comment by ckabalan** [comment]
This is something I would like to do but I don't have a timeline yet. I'm not as familiar with IPv6, are there any IPv6-specific features you would look for? Otherwise just equivalent functionality but with IPv6 addresses...

**Comment by theotherguy2175** [comment]
Totally understandable. For v6 I suppose you could to split all the way
from /1 to /128, but in reality, doing the splitting in increments of 16 or
4-8 would actually be better for screen real estate in my opinion. That's
what we use mostly anyways. It's rare they are divided smaller than that.

I took a brief look at your code. And it looks like the main HTML page does
the input validation for v4 via regex. Something similar could be done for
v6 with this:

^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))(?=\/(1[0-2][0-8]|[1-9][0-9]?|[1-9])$).*$

And then just calculate the usable host range by using this simple formula:
 2^(128 - [mask])

**Comment by jasontally** [comment]
Yes v6 would be awesome! limiting splits to levels divisible by 4 and not allowing splits past /64 would be good but just plain v6 support would be a big step up from the options we have today.

**Comment by My-Random-Thoughts** [comment]
I use the following for IPv6 Regex:

`^((([0-9a-f]{1,4}:){7}([0-9a-f]{1,4}|:))|(([0-9a-f]{1,4}:){6}(:[0-9a-f]{1,4}|((25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])|:))|(([0-9a-f]{1,4}:){5}(((:[0-9a-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])|:))|(([0-9a-f]{1,4}:){4}(((:[0-9a-f]{1,4}){1,3})|((:[0-9a-f]{1,4})?:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9]))|:))|(([0-9a-f]{1,4}:){3}(((:[0-9a-f]{1,4}){1,4})|((:[0-9a-f]{1,4}){0,2}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9]))|:))|(([0-9a-f]{1,4}:){2}(((:[0-9a-f]{1,4}){1,5})|((:[0-9a-f]{1,4}){0,3}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9]))|:))|(([0-9a-f]{1,4}:){1}(((:[0-9a-f]{1,4}){1,6})|((:[0-9a-f]{1,4}){0,4}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9]))|:))|(:(((:[0-9a-f]{1,4}){1,7})|((:[0-9a-f]{1,4}){0,5}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4]\d|1\d\d|[1-9]?[0-9]))|:)))$`

It was taken from this code: https://www.powershellgallery.com/packages/IPv6Regex/1.1.1/Content/IPv6Regex.psm1
I have found it to be more reliable and covers more test cases.

**Comment by unquietwiki** [comment]
Just saw this on TLDR, and yeah, v6 support would be nice. And yeah, you might actually need to split past /64 (some virtual services use /80 or /112).

**Comment by Justman100** [comment]
@ckabalan

You can also look [here](https://subnettingpractice.com/ipv6-subnet-calculator.html)...

**Comment by Justman100** [comment]
But don't forget the same functions suggested by me in #31

**Comment by jasontally** [comment]
I ended up making a similar IPv6 tool. You can find it at [https://v6plan.jasontally.com](https://v6plan.jasontally.com)

**Comment by Justman100** [comment]
> I ended up making a similar IPv6 tool. You can find it at https://v6plan.jasontally.com

The source?

**Comment by jasontally** [comment]
There is a link on the page, but it is also here: [https://github.com/jasontally/IPv6-Plan](https://github.com/jasontally/IPv6-Plan)

**Comment by alsyundawy** [comment]
https://alsyundawy.github.io/visualsubnetcalc/

**Comment by Justman100** [comment]
> https://alsyundawy.github.io/visualsubnetcalc/

👀👀👀
